### ISSUE TITLE
SA1513 fires in switch statements

### ISSUE THREAD

**Issue by Nihlus** [DotNetAnalyzers/StyleCopAnalyzers#2509]
SA1513 fires erroneously for switch statements with enforced braces. A block of code structured in the following manner

```cs
int i = 0;
switch (i)
{
    case 1:
    {
        // Thing
        break;
    }
    case 2:
    {
        // Other thing
        break;
    }
    default:
    {
        break;
    }
}
```
will raise two instances of the inspection at lines 8 and 13, respectively. I would assume that this is not intended behaviour, and that no instances should be raised. If this is intended, it would be a good candidate for addition into the json configuration, as it is a common pattern for switch statements and introducing blank lines between the cases would go against this.

**Comment by vweijsters** [comment]
This is indeed behavior by design. I'll mark this request as an enhancement proposal.

**Comment by Nihlus** [comment]
Super :)

**Comment by sharwell** [comment]
I believe this is a good candidate for implementation separately from this project (after dotnet/roslyn#20242 gets designed/implemented).

**Comment by ghostelet** [comment]
Hi,

Here's another case where this issue happens (assuming `Foo` is an `enum`) using .NET Core 3.1 or .NET 5:

        internal static int FooTest(Foo foo, int a)
            => foo switch
            {
                Foo.Bar => 1,
                Foo.Buzz => 2,
                _ => throw new InvalidOperationException(),
            } << a;

forces me to write:

        internal static int FooTest(Foo foo, int a)
            => foo switch
            {
                Foo.Bar => 1,
                Foo.Buzz => 2,
                _ => throw new InvalidOperationException(),
            }

            << a;

**Comment by sharwell** [comment]
@ghostelet that's a different bug. Can you file it separately?

**Comment by sharwell** [comment]
Closing this one as **wontfix** in favor of external implementation of a different role based on https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2509#issuecomment-319524115. I believe an implementation of one alternative is already underway in dotnet/roslyn-analyzers.

**Comment by ghostelet** [comment]
@sharwell Done : https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3288
